src/main/java/org/jabref/model/openoffice/rangesort/RangeSort.java

```diff
@@ -0,0 +1,95 @@
+package org.jabref.model.openoffice.rangesort;
+
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.jabref.model.openoffice.uno.UnoCast;
+import org.jabref.model.openoffice.uno.UnoTextRange;
+
+import com.sun.star.text.XText;
+import com.sun.star.text.XTextRangeCompare;
+
+public class RangeSort {
+
+    /*
+     * Sort within a partition
+     */
+
+    public static class HolderComparatorWithinPartition implements Comparator<RangeHolder> {
+
+        XTextRangeCompare cmp;
```

Comment:
ok, `private final XTextRangeCompare cmp;`

HolderComparatorWithinPartition itself could be private as well.

Changed class comment (on to: HolderComparatorWithinPartition)
```
    /**
     * Compare two RangeHolders (using RangeHolder.getRange()) within an XText.
     *
     * Note: since we only look at the ranges, this comparison is generally not consistent with
     * `equals` on the RangeHolders. Probably should not be used for key comparison in
     * TreeMap&lt;RangeHolder&gt; or Set&lt;RangeHolder&gt;
     *
     */
```

Note: `TreeMap&lt;RangeHolder&gt; or Set&lt;RangeHolder&gt;` is hard to read, but I believe javadoc `/** */` expects it.
Is there a way to make the comment both readable and javadoc-compatible?

Maybe, like `/*  Object.equal */`,  the content of the "Note:" part it is expected to be well-known among java users,
thus superfluous.

Reply:
> HolderComparatorWithinPartition itself could be private as well.

Fair enough. If you aren't going to use it in a later PR.

> Note: TreeMap&lt;RangeHolder&gt; or Set&lt;RangeHolder&gt; is hard to read, but I believe javadoc /** */ expects it.

What about `{@code TreeMap<RangeHolder>} or {@code Set<RangeHolder>}`?

> Maybe, like /* Object.equal */, the content of the "Note:" part it is expected to be well-known among java users,
thus superfluous.

Kind of, but with `/* Object.equal */`, you are already saying that you are `@Override` an inherited method. I am also not sure that the concept of consistency with equals is that well-known X)

Perhaps say that the natural ordering isn't consistent with equals? I don't think it is likely to be an issue, and depending on what you are extending with `RangeHolder` it might not be true.